compute_field_after_install/models/recompute_field.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2017 Akretion (http://www.akretion.com).
+# @author Sébastien BEAU <sebastien.beau@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+import logging
+
+from openerp import api, fields, models
+from openerp.exceptions import Warning as UserError
+from openerp.tools.translate import _
+
+from odoo.tools import config
+
+_logger = logging.getLogger(__name__)
+
+
+class RecomputeField(models.Model):
+    _name = "recompute.field"
+    _description = "Recompute Field"
+
+    model = fields.Char(required=True)
+    field = fields.Char(required=True)
+    last_id = fields.Integer(
+        string="Last ID", help="Last record ID on which computing have been executed"
+    )
+    step = fields.Integer(required=True, default=1000, help="Recomputing batch size.")
+    state = fields.Selection(
+        [
+            ("todo", "Todo"),
+            ("done", "Done"),
+        ]
+    )
+
+    @api.model
+    def _run_all(self):
+        return self.search([("state", "=", "todo")]).run()
+
+    def run(self):
+        for task in self:
+            cursor = self.env.cr
+            offset = 0
+            model = self.env[task.model]
+            if task.last_id:
+                domain = [("id", ">", task.last_id)]
+            else:
+                domain = []
+            if task.step <= 0:
+                raise UserError(_("Step must be upper than 0"))
+            else:
+                limit = task.step
+            while True:
+                _logger.info(
+                    "Recompute field %s for model %s in background. Offset %s",
+                    task.field,
+                    task.model,
+                    offset,
+                )
+                records = model.search(domain, limit=limit, offset=offset, order="id")
```

Comment:
```suggestion
                records = model.search(domain, limit=limit, offset=offset, order="id desc")
```

I think we should compute first the last created items. Due to @rvalyi remarks, if we try to mark as paid invoices, for sure we will work on recent invoices, not invoices created in 2009. (well, it can occures, but less frequently).

This proposal should come with other changes, regarding last_id, etc..

Reply:
I agree, I prefer newest IDs to be recomputed first, rather than the oldest IDs.